exporter/datadogexporter/internal/metrics/transform_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/metrics"
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/testutils"
+)
+
+func TestYAMLIsValid(t *testing.T) {
```

Comment:
I would also explicitly test that the configuration unmarshals correctly into the metrics transform processor struct. We are kind of doing this implicitly because of the init function, but if we ever change that we could inadvertedly reduce coverage, so I would rather do it here too.

Reply:
I'll wait for your response on the [other comment](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/13980/files#r970527550) and adjust the test correspondingly.